File: skillsync-frontend/src/services/MatchingService.ts
import axios from 'axios';
import { MatchedUserDTO, MatchedProjectDTO } from '../model/Matching';
import Messages from '../messages/Messages';

const MatchingService = {
  findMatchesForProject(projectId: number): Promise<MatchedUserDTO[]> {
    return axios.get(`${Messages.BASE_URL}/matching/project/${projectId}`).then(res => res.data);
  },

  findMatchingProjects(userId: number): Promise<MatchedProjectDTO[]> {
    return axios.get(`${Messages.BASE_URL}/matching/user/${userId}`).then(res => res.data);
  },
};

export default MatchingService;

File: skillsync-backend/src/main/java/com/skillsync/service/MatchingService.java
package com.skillsync.service;

import com.skillsync.dto.MatchedProjectDTO;
import com.skillsync.dto.MatchedUserDTO;
import com.skillsync.utility.SkillSyncException;

import java.util.List;

public interface MatchingService {
    List<MatchedUserDTO> findMatchesForProject(Long projectId) throws SkillSyncException;
    List<MatchedProjectDTO> findMatchingProjects(Long userId) throws SkillSyncException;
}
